src/calc_FD_CG_2DES.c

```diff
@@ -0,0 +1,385 @@
+    read_in_QF_for_FD_CG_2DES(non,Q1,Q2,segments);
+
+    re_doorway   = (float *)calloc(non->tmax*9*segments,sizeof(float));
+    im_doorway   = (float *)calloc(non->tmax*9*segments,sizeof(float));
+    re_window_SE = (float *)calloc(non->tmax*9*segments,sizeof(float));
+    im_window_SE = (float *)calloc(non->tmax*9*segments,sizeof(float));   
+    re_window_GB = (float *)calloc(non->tmax*9*segments,sizeof(float));
+    im_window_GB = (float *)calloc(non->tmax*9*segments,sizeof(float)); 
+    re_window_EA = (float *)calloc(non->tmax*9*segments,sizeof(float));
+    im_window_EA = (float *)calloc(non->tmax*9*segments,sizeof(float)); 
+    P_DA = (float *)calloc(segments*segments,sizeof(float));
+    
+    printf("Performing the FD_CG_2DES calculation.\n"); 
+ 
+    if (!strcmp(non->technique, "FD_CG_2DES") ||  (!strcmp(non->technique, "FD_CG_2DES_doorway"))) {
+        CG_2DES_doorway(non, re_doorway, im_doorway);
+    }  
+     if (!strcmp(non->technique, "FD_CG_2DES") ||  (!strcmp(non->technique, "FD_CG_2DES_window_SE")) ) {
+        CG_2DES_window_SE(non, re_window_SE, im_window_SE); 
+     } 
+      if (!strcmp(non->technique, "FD_CG_2DES") ||  (!strcmp(non->technique, "FD_CG_2DES_window_GB"))){
+        CG_2DES_window_GB(non, re_window_GB, im_window_GB); 
+      }
+  
+      if (!strcmp(non->technique, "FD_CG_2DES") ||  (!strcmp(non->technique, "FD_CG_2DES_window_EA"))){
+        CG_2DES_window_EA(non, re_window_EA, im_window_EA); 
+      }
+
+    /* Call the rate routine routine */
+    if (!strcmp(non->technique, "FD_CG_2DES")||  (!strcmp(non->technique, "FD_CG_full_2DES_segments")) || (!strcmp(non->technique, "FD_CG_2DES_waitingtime"))){
+      CG_2DES_P_DA(non,P_DA,segments);
+        printf("Starting calculation of the 2DES with specific time delay\n");
+        
+        if ((!strcmp(non->technique, "FD_CG_2DES_waitingtime"))){
+            /* Read in absorption, emission and coupling from file if needed */
+	        printf("Calculating spectroscopy from precalculated doorway function, window function\n");
+	        read_doorway_window_from_file(non,"CG_2DES_doorway.dat",im_doorway,re_doorway,non->tmax1);
+            read_doorway_window_from_file(non,"CG_2DES_windows_EA.dat",im_window_EA,re_window_EA,non->tmax1);
+            read_doorway_window_from_file(non,"CG_2DES_windows_GB.dat",im_window_GB,re_window_GB,non->tmax1);
+            read_doorway_window_from_file(non,"CG_2DES_windows_SE.dat",im_window_SE,re_window_SE,non->tmax1);
+            printf("Completed reading pre-calculated data.\n");
+        }
+        
+        call_final_FD_CG_2DES(non,P_DA,segments,re_doorway,im_doorway,
+                              re_window_SE,im_window_SE,
+                              re_window_GB,im_window_GB,
+                              re_window_EA,im_window_EA,
+                              re_2DES, im_2DES, Q1, Q2);
+        /*CG_full_2DES_segments(non,re_doorway,im_doorway,
```

Comment:
This is commented out and just the stuff from the original CG code. I would remove it.